### PR TITLE
Lots of web workflow, C3 and title bar fixes

### LINKED ISSUE #6221: QTPY ESP32C3 import wifi hangs
executing `import wifi` on the qtpy espc2c3  causes the board to hang
```

Adafruit CircuitPython 7.3.0-beta.0 on 2022-04-01; Adafruit QT Py ESP32C3 with ESP32-C3FN4
>>> 
>>> import wifi

```

It has to be power cycled to regain access.

The WiFi works fine on the esp32c3 devkit board.

edited to add: 
The QTPY ESP32C3 wifi  also works with MicroPython v1.18 using the ESP32C3-usb build

### LINKED ISSUE #6653: Web Workflow Serial Terminal screen stops functioning
### CircuitPython version

```python
Adafruit CircuitPython 8.0.0-alpha.1-69-gbc926b041-dirty on 2022-07-26; Adafruit QT Py ESP32C3 with ESP32-C3FN4
Adafruit CircuitPython 8.0.0-alpha.1-69-gbc926b041-dirty on 2022-07-26; Adafruit Feather ESP32S2 with ESP32S2
Adafruit CircuitPython 8.0.0-alpha.1-69-gbc926b041-dirty on 2022-07-26; Adafruit Feather ESP32S3 4MB Flash 2MB PSRAM with ESP32S3

Circuit Python Built from Pull Request #6651
```

### Code/REPL

```python
cmd = ""
while cmd.upper() != "EXIT":
    cmd = input(":>")
```

### Behavior

When each of the boards boots you are able to connect to them over the Web Workflow and open a serial terminal window. On boot the code.py program is running and the serial terminal window can be refreshed without problem.

However refreshing the serial terminal screen at certain points will disconnect the terminal (the C, D and SEND buttons go grey and input to the command line is blocked). The following message is displayed on the browser Dev Tool console:

image

Connecting to REPL via the USB serial port of a board in this state appears to show some of the header bar displayed in the terminal output. If the Workflow serial terminal attempted to display the header string/icons as terminal text that might account for the UTF-8 decode error.

image

Once this error occurs new serial terminal windows will not connect and the existing window continues to error if refreshed. A board hard reset or power cycle appears to be the only way to return the serial terminal back to operation.

### Description

The most consistent way I've been able to reproduce the error is to type exit into the command line and then press the `SEND` button (don't use the enter key). Once the Press any key to enter the REPL message is displayed, refresh the browser screen. Usually refreshing once will cause the error which sometimes takes a minute to occur, but if the refresh works once, pressing refresh a second time will usually trigger the error.

The S3 board seems to throw the error the quickest and in more situations than the others.

### Additional information

I did this testing using builds from PR #6651

### LINKED ISSUE #6655: Can't "Paste" into Web Workflow serial terminal command line
### CircuitPython version

```python
Adafruit CircuitPython 8.0.0-alpha.1-69-gbc926b041-dirty on 2022-07-26; Adafruit Feather ESP32S3 4MB Flash 2MB PSRAM with ESP32S3

Circuit Python Built from Pull Request #6651
```

### Code/REPL

```python
None
```

### Behavior

If you attempt to paste text into the command line of the serial terminal the text is displayed as it's pasted, however pressing enter or the "Send" button only sends a carriage return to the microcontroller

### Description

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by tannewt**
* Fixes #6221 - C3 hang on `import wifi`. Enabling the WiFi PHY was
  disabling USB. Now boards that use it set CONFIG_ESP_PHY_ENABLE_USB
  explicitly.
* Fixes #6655 - Allows pasting into the web serial page. Fixes reading
  more than 0xf bytes at a time.
* Fixes #6653 - Fixes web socket encoding of payloads >125 bytes. Can
  happen when printing a long string.
* Fixes C3 responsiveness when waiting for key to enter REPL. (It
  now correctly stops sleeping.)
* Disables title bar updates when in raw REPL. Related to #6548.
* Adds version to title bar.

### CODE REVIEW

**Comment by RetiredWizard** [comment]
Maybe I messed up my git commands?

     common-hal/alarm/touch/TouchAlarm.c:33:10: fatal error: supervisor/esp_port.h: No such file or directory
      #include "supervisor/esp_port.h"
               ^~~~~~~~~~~~~~~~~~~~~~~

**Comment by tannewt** [comment]
> ```
>  common-hal/alarm/touch/TouchAlarm.c:33:10: fatal error: supervisor/esp_port.h: No such file or directory
>   #include "supervisor/esp_port.h"
>            ^~~~~~~~~~~~~~~~~~~~~~~
> ```
> 
> Just checked the automated build for the PR and they ran into the same thing...

Yup, this is what I get for not building all boards myself. Will fix now.

File: supervisor/shared/serial.c [hunk 1]
```diff
@@ -207,10 +207,17 @@ char serial_read(void) {
 
     #if CIRCUITPY_WEB_WORKFLOW
     if (websocket_available()) {
-        return websocket_read_char();
+        char c = websocket_read_char();
```
Comment (jepler) [inline supervisor/shared/serial.c:210]: getc() can differentiate between EOF and one of the valid values of 'char' by having the return type of "int". However, `websocket_read_char()` returns `char`.  One manpage explains how (f)getc works as follows:

> fgetc()  reads  the next character from stream and returns it as an *unsigned char* cast to an int, or EOF on end of file or error.

**Review by dhalbert — CHANGES_REQUESTED** [review]
I tried the latest build on a Feather ESP32-S2 TFT. I saw the title bar come up when I gave an incorrect wifi password, but when the password was fine, the title bar did not change (!). This was with tio connected, even after pressing reset. I wonder if the title bar string is being sent before the connection has come back up.

**Review by dhalbert — APPROVED** [review]
Thanks for the other fixes! Approving now to get them in; the terminal title bar problem will be fixed later.

**Pull request merged**